File: tests/fixtures/realistic/path_traversal/path_traversal.py
# Phase 13 path-traversal positive (Python).  Flask handler reads
# `request.args.get("name")` (Source) and feeds the value into
# `Path(name).read_text()`.  After paren-strip the chain text becomes
# `Path.read_text`, which matches the new pathlib FILE_IO rule in
# `src/labels/python.rs`.
from flask import request
from pathlib import Path


def handler():
    name = request.args.get("name")
    return Path(name).read_text()

File: tests/fixtures/realistic/path_traversal/path_traversal.rs
// Phase 13 path-traversal positive (Rust).  axum-shape async handler
// reads a header value (Source via the `headers.get` rule in
// `src/labels/rust.rs`), awaits, and feeds the result into
// `tokio::fs::read` (new FILE_IO sink rule).  Mirrors the existing
// `tests/fixtures/realistic/async_await/handler.rs` source shape.
#[allow(unused)]
struct Request;
impl Request {
    fn headers(&self) -> Headers {
        Headers
    }
}
struct Headers;
impl Headers {
    async fn get(&self, _key: &str) -> String {
        String::new()
    }
}

pub async fn handler(req: Request) -> Vec<u8> {
    let path = req.headers().get("X-Path").await;
    tokio::fs::read(path).await.unwrap_or_default()
}
